ocs_ci/utility/utils.py

```diff
@@ -1137,3 +1137,27 @@ def ceph_health_check(namespace=None):
         raise CephHealthException(
             f"Ceph cluster health is not OK. Health: {health}"
         )
+
+
+def get_rook_repo(to_checkout):
+    """
+    Clone and checkout the rook repository to specific branch/commit.
+
+    Args:
+        to_checkout (str): Commit id or branch to checkout on
+
+    """
+    fresh_clone = False
```

Comment:
This can cause stale issues, Rather age of dir can be used to decided freshness.

Reply:
Not sure which stale issues you mean @vasukulkarni ? Can you please elaborate?
Basically if the directory data/rook doesn't exist it means that we don't have the rook repo cloned yet and then we clone it on line 1155.  As we just cloned the repository there is no need to do git fetch --all on line 1159 cause we just cloned the repository so we can directly checkout where we need.

The next run we won't clone the repository and we will just git fetch --all to make sure we have all recent changes fetched so we should be up to date every time I think.

Reply:
@petr-balogh I meant instead of using flag we can use datetime.timedelta to figure out if the dir created is older than 7 days and if so pull the rook files again.

Reply:
Isn't `fresh_clone` just determining whether we `clone` or `fetch` the latest before checking out the branch/commit? In either case we should be getting the latest files so I don't think there's an issue with stale files here. However, is `fresh_clone` even necessary? Seems we can get by with a simple if/else:

```python
if not os.path.isdir(constants.ROOK_REPO_DIR):
    # clone
else:
    # fetch
# checkout
```

Reply:
Actually I realized that if we will have already cloned repo and will just git fetch --all we won't apply the latest changes, so I think that if we will reset --hard to specific branch will be to safest solution.  Also separated this to_commit just for commit id or tag so we can then checkout to the tag or commit once we have fetched latest changes.

Reply:
Would [git pull](https://git-scm.com/docs/git-pull) be more appropriate?

Reply:
If there will be some changes done by user from some reason I thin it can lead to some conflict. With reset --hard you will make sure you are reseted exactly to the same code of origin without need of deal with any conflict or so.

Reply:
Oh I didn't realize that you were doing `git reset --hard origin/{branch}`. I was thinking just `git reset --hard` which would only reset you to your local branch's HEAD which would not update the files, but the way you have it works.